modules/apps/item-selector/item-selector-web/src/test/java/com/liferay/item/selector/web/internal/ItemSelectorImplTest.java

```diff
@@ -106,6 +107,23 @@ public static void setUpClass() {
 		portalUtil.setPortal(new PortalImpl());
 	}
 
+	@After
+	public void tearDown() {
+		if ((_flickrItemSelectorCriterionHandlerServiceRegistration != null) ||
+			(_mediaItemSelectorCriterionHandlerServiceRegistration != null)) {
+
+			_itemSelectorImpl.deactivate();
+
+			_flickrItemSelectorCriterionHandlerServiceRegistration.unregister();
+
+			_flickrItemSelectorCriterionHandlerServiceRegistration = null;
```

Comment:
@W2jx I doubt if it's needed to set this to null. Please take a look at how JUnit 4 works to confirm.

Reply:
Hi @dantewang , The reason why I make them null is that test case 'testGetItemSelectorParameters' does not need to execute method `_setUpItemSelectionCriterionHandlers`, so I need a flag to determine whether method `_itemSelectorImpl.deactivate() ` should be executed.